bitnami/fluent-bit/templates/deployment.yaml

```diff
@@ -0,0 +1,153 @@
+{{- if not .Values.useDaemonset }}
+apiVersion: {{ include "common.capabilities.deployment.apiVersion" . }}
+kind: Deployment
+metadata:
+  name:  {{ include "fluentBit.fullname" . }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    app.kubernetes.io/part-of: fluent-bit
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+    app.kubernetes.io/part-of: fluent-bit
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  selector:
+    matchLabels: {{- include "common.labels.matchLabels" . | nindent 6 }}
+  replicas: {{ .Values.replicaCount }}
```

Comment:
If it has an HPA we have checks for not setting the replicas. Check other charts but I remember that it was conflicting setting replicas and the HPA at the same time

Reply:
[1f6b5d2](https://github.com/bitnami/charts/pull/15453/commits/1f6b5d22bc63aaeb4e64fc08d855e8e03438fcf3)

Reply:
Have to mention that many charts with hpa's don't have this check: apache, kong, argo-cd, aspnet-core... Basically all the ones that I've checked. Anyway I did just:

```
  {{- if not .Values.autoscaling.hpa.enabled }}
  replicas: {{ .Values.replicaCount }}
  {{- end }}
```

In this case. But we may need to check that is not conflicting in all charts we have using hpa's and not implementing this.